base/views.py

```diff
@@ -28,8 +48,20 @@ def signup(request):
         
         try:
             user = User.objects.create_user(username=email, email=email, password=password, cnic=cnic, dob=dob, phone=phone, address=address)
+            
+            token = get_random_string(length=32)
+            user.verification_token = token
             user.save()
-            messages.success(request, "Account created successfully")
+            
+            verification_url = request.build_absolute_uri(reverse('verify_email', kwargs={'token': token}))
+            send_mail(
+                'Verify your email address',
+                f'Click the link to verify your email: {verification_url}',
+                settings.DEFAULT_FROM_EMAIL,
+                [email],
+                fail_silently=False,
+            )
+            messages.success(request, "Account created successfully, Please check your email for verification")
             return redirect('login')
         except IntegrityError:
             messages.error(request, "That email is already taken.")
```

Comment:
return proper exception generated instead of hardcoding `That email is already taken.`